test_install/CMakeLists.txt

```diff
@@ -33,29 +43,55 @@ if(GINKGO_BUILD_CUDA)
             ginkgo_switch_to_windows_static("CUDA")
         endif()
     endif()
-    add_executable(test_install_cuda test_install_cuda.cu)
+    set_source_files_properties(test_install.cpp PROPERTIES LANGUAGE CUDA)
+    add_executable(test_install_cuda test_install.cpp)
+    target_compile_options(test_install_cuda
+        PRIVATE "$<$<COMPILE_LANGUAGE:CUDA>:${GINKGO_CUDA_ARCH_FLAGS}>")
+    target_compile_definitions(test_install_cuda PRIVATE HAS_CUDA=1)
     target_link_libraries(test_install_cuda PRIVATE Ginkgo::ginkgo)
 endif()
 
 if(GINKGO_BUILD_HIP)
     find_package(HIP REQUIRED)
 
-    set_source_files_properties(test_install.hip.cpp PROPERTIES HIP_SOURCE_PROPERTY_FORMAT TRUE)
+    set_source_files_properties(test_install.cpp PROPERTIES HIP_SOURCE_PROPERTY_FORMAT TRUE)
     # The library was compiled with `-fPIC` when in shared library mode
     if (GINKGO_BUILD_SHARED_LIBS)
         set (GINKGO_PIC_OPTION "-fPIC")
         set (GINKGO_CUDA_PIC_OPTION "-Xcompiler '-fPIC'")
     endif()
-    hip_add_executable(test_install_hip test_install.hip.cpp
+    if (CMAKE_CUDA_COMPILER_ID STREQUAL "NVIDIA")
+        set(TESTINSTALL_CUDA_ARCH_FLAGS "${GINKGO_CUDA_ARCH_FLAGS}")
```

Comment:
Do you want to use the minimal flags on test_install?
otherwise, maybe reuse `GINKGO_HIP_NVCC_OPTIONS`

Reply:
There is a bug here. `hipcc` always uses `nvcc`, even if you set `CMAKE_CUDA_COMPILER` to be clang. If you used `clang` for compiling Ginkgo, `${GINKGO_CUDA_ARCH_FLAGS}` will work only with `clang` and `hipcc` will fail to compile because the flags are for `clang` and not `nvcc`. On the other hand, this is such a corner case I think it's not worth the trouble of exporting the CudaArchitectureSelector to the test_install.